rplugin/python3/deoplete/source/buffer.py

```diff
@@ -29,8 +29,12 @@ def on_event(self, context):
     def gather_candidates(self, context):
         self.__make_cache(context)
 
-        buffers = [x['candidates'] for x in self.__buffers.values()
-                   if x['filetype'] in context['filetypes']]
+        buffer_values = self.__buffers.values()
+        if context['vars'].get('require_same_filetype', True):
+          buffer_values = [x for x in buffer_values
```

Comment:
Sorry, in the snippet I posted above, I set the default for `require_same_filetype` to `False`, but I think it makes more sense if it's `True` by default.